hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -1009,6 +1000,20 @@ public HdfsFileStatus getFileInfo(String src) throws IOException {
     return ret;
   }
 
+  public RemoteResult<RemoteLocation, HdfsFileStatus> getFileRemoteResult(String path)
+      throws IOException {
+    rpcServer.checkOperation(NameNode.OperationCategory.READ);
+
+    final List<RemoteLocation> locations = rpcServer.getLocationsForPath(path, false, false);
+    RemoteMethod method =
+        new RemoteMethod("getFileInfo", new Class<?>[] {String.class}, new RemoteParam());
+    // Check for file information sequentially
+    RemoteResult<RemoteLocation, HdfsFileStatus> result =
```

Comment:
If `locations` only contains one namespace, we can returns this namespace directly instead of getting the namespace through `getFileInfo`, right?

Reply:
This may not be  true. Even locations only contains one namespace, it still cannot decide whether the file exists or not. So getFileInfo is better to execute at least once. Or it will send to namenode and  namenode throw file is not found.

Reply:
RBF is simply responsible for locating the downstream namespace and then proxying the request.
So if the input path is only mounted to one namespace, RBF only needs to proxy it directly.  RBF does not need to check if the file exists in this only one downstream namespace, right?

Reply:
OK